### PR TITLE
Update building-docker-images.md

### LINKED ISSUE #3124: Docker Version Latest Stable and Edge Not Supported
# Locations
https://circleci.com/docs/2.0/building-docker-images/#docker-version

CircleCI supports multiple versions of Docker and defaults to using docker-17.09.0-ce. Consult the Stable releases or Edge releases for the full list of supported versions.

Note: The version key is not currently supported on CircleCI installed in your private cloud or datacenter. Contact your system administrator for information about the Docker version installed in your remote Docker environment.

# Details
Documentation lists all 'stable' and 'edge' versions listed in the Docker upstream repository as available but this is incorrect. 

These are clearly not authoritative for CircleCI in entirety:
https://download.docker.com/linux/static/stable/x86_64/
https://download.docker.com/linux/static/edge/x86_64/

docker-18.09.2 is clearly listed but if you attempt to specify you get an error message that it's not supported in CircleCI.

'image docker-18.09.2 is not supported'

### PR DESCRIPTION
**Pull request by michelle-luna**
fixes #3124

# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/building-docker-images.md [hunk 1]
```diff
@@ -118,7 +118,30 @@ If your job requires a specific docker version, you can set it as a `version` at
           version: 18.06.0-ce
 ```
 
-CircleCI supports multiple versions of Docker and defaults to using `docker-17.09.0-ce`. Consult the [Stable releases](https://download.docker.com/linux/static/stable/x86_64/) or [Edge releases](https://download.docker.com/linux/static/edge/x86_64/) for the full list of supported versions.
+CircleCI supports multiple versions of Docker and defaults to using `docker-17.09.0-ce`. Following are the supported stable and edge versions:
```
Comment (glenjamin) [inline jekyll/_cci2/building-docker-images.md]: We should drop the `docker-` prefix from all of these examples, that’s not needed when the user specifies the version in their config - i’m not sure why it was there before either.

I’ll double-check this tomorrow.

Comment (michelle-luna) [reply jekyll/_cci2/building-docker-images.md]: thanks, I removed those (the example didn't include docker- prefix so I'm sure you are correct

File: jekyll/_cci2/building-docker-images.md [hunk 2]
```diff
@@ -118,7 +118,30 @@ If your job requires a specific docker version, you can set it as a `version` at
...
 ```
 
-CircleCI supports multiple versions of Docker and defaults to using `docker-17.09.0-ce`. Consult the [Stable releases](https://download.docker.com/linux/static/stable/x86_64/) or [Edge releases](https://download.docker.com/linux/static/edge/x86_64/) for the full list of supported versions.
+CircleCI supports multiple versions of Docker and defaults to using `17.09.0-ce`. Following are the supported stable and edge versions:
+
+- `17.03.0-ce`
+- `17.05.0-ce`
+- `17.06.0-ce`
+- `17.06.1-ce`
+- `d17.07.0-ce`
```
Comment (glenjamin) [inline jekyll/_cci2/building-docker-images.md]: You have a stray `d` here.

File: jekyll/_cci2/building-docker-images.md [hunk 3]
```diff
@@ -118,7 +118,30 @@ If your job requires a specific docker version, you can set it as a `version` at
...
+- `17.12.0-ce`
+- `17.12.1-ce`
+- `18.01.0-ce`
+- `18.02.0-ce`
+- `18.03.0-ce`
+- `18.03.1-ce`
+- `18.04.0-ce`
+- `18.05.0-ce`
+- `18.06.0-ce`
+- `18.09.3-ce`
```
Comment (glenjamin) [inline jekyll/_cci2/building-docker-images.md]: We've updated this one to drop the `-ce` suffix, because docker started dropping this since `18.09.0`.

**Pull request merged**